### PR TITLE
Prepare for .NET 8

### PR DESCRIPTION
**Pull request by martincostello**
Port various changes from #1144:

- Simplify NuGet package version management.
- Simplify `TimeProviderExtensions`.
- Enable .NET analyzers.
- Remove `MockTimeProvider`.
- Minor code formatting clean-ups.

Also updates the Cake dependencies to their latest versions.

### CODE REVIEW

File: test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs [hunk 1]
```diff
@@ -224,7 +224,7 @@ public async Task Execute_EnsureCancellationTokenRegistrationNotExecutedOnSynchr
             },
             cts.Token);
         }
-        catch (OperationCanceledException)
+        catch (TaskCanceledException)
```
Comment (martintmk) [inline test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs:227]: Is this change necessary?

Comment (martincostello) [reply test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs:227]: In the .NET 8 branch at least, the tests were failing without this change.

**Review by martintmk — APPROVED** [review]
lgtm

File: test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs (hunk 1 above)
Comment (martintmk) [reply test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs:227]: This is super odd, the `TaskCanceledException` just derives from `OperationCanceledException`

Comment (martincostello) [reply test/Polly.Core.Tests/Timeout/TimeoutResilienceStrategyTests.cs:227]: Let me just revert it locally and double-check in case I got confused somewhere...

File: src/Polly.Core/Utils/TimeProviderExtensions.cs [hunk 2]
```diff
@@ -23,33 +23,15 @@ public static Task DelayAsync(this TimeProvider timeProvider, TimeSpan delay, Re
...
-                Thread.Sleep(delay);
-            }
-
-            return Task.CompletedTask;
-        }
-        else
-        {
-            if (context.IsSynchronous)
-            {
 #pragma warning disable CA1849 // For synchronous scenarios we want to return completed task
```
[the remark this answers is not in the document]
Comment (martintmk) [reply src/Polly.Core/Utils/TimeProviderExtensions.cs:28]: and that we are aware of sync-over-async :)

[the remark this answers is not in the document]
Comment (martincostello) [reply src/Polly.Core/Utils/TimeProviderExtensions.cs]: Added - let me know if it makes sense 😄

[the remark this answers is not in the document]
Comment (martintmk) [reply src/Polly.Core/Utils/TimeProviderExtensions.cs]: Looks good :)
